### PR TITLE
Add flag to force sequential block for ifThenElse

### PR DESCRIPTION
**Pull request by christiaanb**
Work-around for #1677

### CODE REVIEW

**Comment by alex-mckenna** [comment]
`TernaryOpt` is quite specific, what about a more generic name like `PreferSequential` so the flag can be reused if there are any similar issues in the future where comb. / seq. versions have different semantics? Although documenting such a flag would be more difficult - _it's sequential if it might need to be_...

File: Clash.hs [hunk 1]
```diff
@@ -33,17 +33,17 @@ import Util (OverridingBool(..))
 genSystemVerilog
   :: String
   -> IO ()
-genSystemVerilog = doHDL (initBackend WORD_SIZE_IN_BITS HDLSYN True PreserveCase Nothing (AggressiveXOptBB False) :: SystemVerilogState)
+genSystemVerilog = doHDL (initBackend WORD_SIZE_IN_BITS HDLSYN True PreserveCase Nothing (AggressiveXOptBB False) (TernaryOpt True) :: SystemVerilogState)
```
Comment (martijnbastiaan) [inline Clash.hs:36]: I think this has reached a breaking point. Could we refactor this into a record?

**Review by martijnbastiaan — CHANGES_REQUESTED** [review]

File: Clash.hs (hunk 1 above)
Comment (alex-mckenna) [reply Clash.hs:36]: Did someone say _refactor this_ :sparkles:

Comment (martijnbastiaan) [reply Clash.hs:36]: :tada:

**Comment by martijnbastiaan** [comment]
Given the conflicts and activity level I'm going to close this PR. Feel free to reopen if you pick up work again.

**Pull request closed without merging**